internal/istio/virtualservice_test.go

```diff
@@ -161,58 +193,109 @@ func Test_NewVirtualService_SetsCorrectHosts(t *testing.T) {
+	watcher := setupWatcher()
+	namespace := setupNamespace()
+	gateways := setupGateways()
 	watcher.Spec.ServiceInfo = v1beta2.Service{}
 
-	vs, err := istio.NewVirtualService(namespace, watcher, gateways)
+	vs, err := vss.NewVirtualService(namespace, watcher, gateways)
 
 	assert.Nil(t, vs)
 	require.ErrorIs(t, err, istio.ErrInvalidArgument)
 	assert.Contains(t, err.Error(), "unable to construct httpRoute from watcher")
 }
 
 func Test_NewVirtualService_SetsAHttpRoute(t *testing.T) {
-	watcher := getSimpleWatcher()
-	namespace := getSimpleNamespace()
-	gateways := getSimpleGateways()
+	vss := setupVirtualServiceService(t)
+	watcher := setupWatcher()
+	namespace := setupNamespace()
+	gateways := setupGateways()
 
-	vs, err := istio.NewVirtualService(namespace, watcher, gateways)
+	vs, err := vss.NewVirtualService(namespace, watcher, gateways)
 
 	require.NoError(t, err)
 	assert.Len(t, vs.Spec.GetHttp(), 1)
 }
 
-func getSimpleWatcher() *v1beta2.Watcher {
+func Test_NewVirtualService_ReturnsError_WhenFailingToAddOwnerReference(t *testing.T) {
+	vss, _ := istio.NewVirtualServiceService(machineryruntime.NewScheme()) // does not provide needed Watcher scheme
+	watcher := setupWatcher()
+	namespace := setupNamespace()
+	gateways := setupGateways()
+
+	vs, err := vss.NewVirtualService(namespace, watcher, gateways)
+
+	require.Nil(t, vs)
+	require.ErrorIs(t, err, istio.ErrFailedToAddOwnerReference)
+}
+
+func Test_NewVirtualService_SetsOwnerReference(t *testing.T) {
+	vss := setupVirtualServiceService(t)
+	watcher := setupWatcher()
+	namespace := setupNamespace()
+	gateways := setupGateways()
+	expectedOwnerReference := getOwnerReference(watcher)
+
+	vs, err := vss.NewVirtualService(namespace, watcher, gateways)
+
+	require.NoError(t, err)
+	ownerReferences := vs.ObjectMeta.GetOwnerReferences()
+	assert.Len(t, ownerReferences, 1)
+	assert.Equal(t, expectedOwnerReference, ownerReferences[0])
+}
+
+func setupVirtualServiceService(t *testing.T) *istio.VirtualServiceService {
+	t.Helper()
+
+	vss, _ := istio.NewVirtualServiceService(setupScheme(t))
+	return vss
+}
+
+func setupScheme(t *testing.T) *machineryruntime.Scheme {
+	t.Helper()
+
+	scheme := machineryruntime.NewScheme()
+	if err := api.AddToScheme(scheme); err != nil {
+		assert.Fail(t, "failed to setup scheme")
+	}
+
+	return scheme
+}
+
+func setupWatcher() *v1beta2.Watcher {
 	watcher := &v1beta2.Watcher{}
 
 	watcher.SetName("watcher-name")
-	watcher.SetNamespace(getSimpleNamespace())
+	watcher.SetNamespace(setupNamespace())
 
 	watcher.Labels = map[string]string{
 		shared.ManagedBy: "some-manager",
 	}
 
 	watcher.Spec.ServiceInfo = v1beta2.Service{
 		Name:      "service-name",
-		Namespace: getSimpleNamespace(),
+		Namespace: setupNamespace(),
 		Port:      4711,
 	}
 
+	watcher.APIVersion = "operator.kyma-project.io/v1beta2"
+	watcher.Kind = "Watcher"
+	watcher.UID = uuid.NewUUID()
+
 	return watcher
 }
 
-func getSimpleNamespace() string {
+func setupNamespace() string {
 	return "bar"
```

Comment:
Also this could be a const? It's not creating a v1.Namespace, but using a function indicates it.

Reply:
Done as part of introducing the watcher builder https://github.com/kyma-project/lifecycle-manager/pull/1349/commits/016f9609061df109e3ad4c9a4fa60949ab981009